vmm/src/lib.rs

```diff
@@ -997,7 +1079,14 @@ impl Vmm {
         };
 
         if send_data_migration.local {
-            vm.send_memory_fds(&mut socket)?;
+            match &mut socket {
+                SocketStream::Unix(unix_socket) => {
+                    vm.send_memory_fds(unix_socket)?;
+                }
+                SocketStream::Tcp(tcp_socket) => {
+                    vm.send_memory_fds_tcp(tcp_socket)?;
```

Comment:
The purpose of `--local` is to not need to read/write the memory and instead use SCM_RIGHTS to move the fd across the process boundaries. `--local` makes no sense with TCP and should generate an error.

Reply:
You are right. I have modified the code as you suggested to ensure that when using TCP, if the user attempts to use the --local option, they will get an error message:)

Thanks